### DISCUSSION TITLE
should NOT consider HK(Hong Kong), MO(Macau), TW(Taiwan) as countries

### DISCUSSION THREAD

**Discussion by jas0nt** [dr5hn/countries-states-cities-database#608] [Ideas]
they are part of China

**Comment by dr5hn** [comment]
@jas0nt One of the users said, "Hong Kong is a country and also a city." https://github.com/dr5hn/countries-states-cities-database/issues/78

So bit confused now 😕 

Please help.

**Comment by jas0nt** [comment]
According to [wikipedia](https://en.wikipedia.org/wiki/Hong_Kong): "Hong Kong is a city and special administrative region of China".

Some places classify Hong Kong as a "country/region", but definitely not a "country".

**Comment by spekulatius** [comment]
One should study the history of Taiwan and maybe even take a trip to Taiwan. Then this question is answered ;)

**Comment by Ocean-Blue** [comment]
About Taiwan, you can watch this video, starting from 1:50
https://www.youtube.com/watch?v=zWdPzWHdnFE&t=181s

You can consider Taiwan vs. China like North/South Korea. The world is often confused because these 2 countries have very similar official country name, just like Korea.
Taiwan's formal country name is "Republic of China". China's formal country name is "People's Republic of China"

**Comment by greg-ynx** [comment]
Since recent events I would say that Macau and Hong Kong are **NOT** countries anymore. They were like "cities states" before but now they are administrative regions of the People's Republic of China. However, Taiwan is a country recognized by the UN. Check this out here : [ℹ️](https://en.wikipedia.org/wiki/List_of_sovereign_states)

**Comment by icopp** [comment]
I feel like the best thing to do here would be to follow the lead of ISO, which for country code uses deprecation warnings (but doesn't remove from the dataset) for countries/states that no longer exist at all, and uses an `independent` flag for cases where an entity is no longer self-governed but still has an ISO code assigned to it.

**Comment by dr5hn** [comment]
Hi,

Thank you all for bringing up the discussion regarding the classification of Hong Kong and Macau within the countries, states, cities database. Based on extensive research and considering various factors, including their administrative status, it has been determined that aligning Hong Kong and Macau as special administrative regions (S.A.R) of China within the countries table is the most accurate representation.

This approach takes into account the official status of Hong Kong and Macau as special administrative regions under the sovereignty of China. While there may be different perspectives and interpretations on their classification, the database aims to reflect the administrative structure as recognized by international standards.

------

Also, Thank you for raising the question about the inclusion of Taiwan within the countries, states, cities database. The classification of Taiwan is indeed a complex matter with various perspectives and interpretations. While Taiwan is a self-governing entity with its own government, it is also recognized as part of the People's Republic of China by some countries and international organizations.

As a database, our goal is to provide accurate and relevant information based on widely recognized standards and conventions. Considering the unique status of Taiwan, it has been included as a separate entry within the countries table, identified as the "Republic of China" (official country name). This is to reflect the distinct political and administrative entity that Taiwan represents.

We understand that there may be differing views on the classification of Taiwan, and we appreciate your insights and perspectives on this matter. The inclusion of Taiwan as a separate entry is not intended to promote any particular political stance but rather to ensure that users have access to information that reflects the diverse and complex nature of global geography.

We appreciate your input and the valuable insights shared in this discussion. If you have any further questions or suggestions, please feel free to contribute. Your feedback helps improve the accuracy and relevance of the database. Thank you for your understanding.
